storage/storage.go

```diff
@@ -729,6 +753,70 @@ func (storage DBStorage) WriteReportForCluster(
 	return err
 }
 
+// WriteRecommendationsForCluster writes hitting rules in received report for selected cluster
+func (storage DBStorage) WriteRecommendationsForCluster(
+	clusterName types.ClusterName,
+	stringReport types.ClusterReport,
+) (err error) {
+	if storage.dbDriverType != types.DBDriverSQLite3 && storage.dbDriverType != types.DBDriverPostgres {
+		return fmt.Errorf("writing report with DB %v is not supported", storage.dbDriverType)
+	}
+
+	// Begin a new transaction.
+	tx, err := storage.connection.Begin()
+	if err != nil {
+		return err
+	}
+
+	var deletedRows int
+	var insertedRows int
+
+	err = func(tx *sql.Tx) error {
+
+		// Delete current recommendations for the cluster
+		result, err := tx.Exec(
+			"DELETE FROM recommendations WHERE cluster_id = $1;", clusterName)
+		err = types.ConvertDBError(err, []interface{}{clusterName})
```

Comment:
Not sure this is needed with the DELETE operation, but used it just in case

Reply:
hmm you want to do `UPSERT` right? Probably better to do INSERT with ON CONFLICT clause?

Reply:
I didn't see it that way. I thought for each report we would "reset" the corresponding clusters, so upsert could leave behind some unwanted entries. But maybe I didn't understand how the recommendations would work

Reply:
Ah you are correct @epapbak